Lib/fontTools/designspaceLib/__init__.py

```diff
@@ -718,7 +717,10 @@ def readRules(self):
                 ruleObject.conditions.append(cd)
             for subElement in ruleElement.findall('.sub'):
                 a = subElement.attrib['name']
-                b = subElement.attrib['byname']
+                if self.toolVersion < 4:
```

Comment:
no problem!
the issue of "byname" vs "with" is a bit complicated, because the spec was saying on thing ("byname") whereas the only implementation (designSpaceDocument == fontTools.designspaceLib) was actually writing something else ("with").
So I changed the implementation to match the spec... So which one was "right" then?

Also, I wonder, are there actual designspaces out there that we wish to support that use either "with" or "byname"? Since the feature was marked as experimental, I think we don't need to be concerned with changing it as we see fit.

Reply:
So you'd rather not check for `byname`?

Reply:
yeah, that's what I was suggesting. Let's just read/write using `width` and change the spec accordingly.
The change in the designspace document `format` attribute (from an integer to a `MAJOR.MINOR` version string) in itself warrants a version bump to 4.0, since it would break compatibility with readers that assumed format was an integer.

Reply:
I also think format should be stored as a string, it's not really a float.

Reply:
Ok. I have some changes here. Is there a way I can show you what I have?

Reply:
I pushed these changes.

Reply:
where did you push them? I see you created a new branch "designspacelib-conditionset-spec" then deleted it. It's ok if you push directly to this PR branch, it's marked as WIP.